### DISCUSSION TITLE
SSL routines - wrong version number

### DISCUSSION THREAD

**Discussion by Mato816** [canarytrace/documentation#188] [Listener]
Hi,

i have problem with configuration for listener.
In YAML CronJob we set:

- SMTP_SERVER
- SMTP_PORT
- SMTP_USER
- SMTP_PASS

After running CronJob we got following error message:

Error: {"library":"SSL routines","reason":"wrong version number","code":"ESOCKET","command":"CONN"}
at /opt/canary-listener/reporters/email.js:116:13
at /opt/canary-listener/node_modules/nodemailer/lib/mailer/index.js:226:21
at SMTPConnection.<anonymous> (/opt/canary-listener/node_modules/nodemailer/lib/smtp-transport/index.js:175:24)
at Object.onceWrapper (node:events:628:26)
at SMTPConnection.emit (node:events:513:28)
at SMTPConnection._onError (/opt/canary-listener/node_modules/nodemailer/lib/smtp-connection/index.js:764:14)
at SMTPConnection._onSocketError (/opt/canary-listener/node_modules/nodemailer/lib/smtp-connection/index.js:190:45)
at TLSSocket.emit (node:events:513:28)
at emitErrorNT (node:internal/streams/destroy:151:8)
at emitErrorCloseNT (node:internal/streams/destroy:116:3)
Node.js v18.7.0
Listener agent: end of analysis

Can you help me please with this issue?
Thank you.

**Comment by rdpanek** [comment]
I am sorry for a later answer.

The error you mentioned is related to an issue with SSL/TLS encryption during the establishment of a socket connection. This error is usually caused by a mismatch between the SSL/TLS version you are using and the version supported by the server you are trying to connect to.

There are several possible causes for this error:

Incorrect SSL/TLS version: If your client program or library is using a specific SSL/TLS version that the server does not support, it can result in this error. It's possible that the server requires a newer or older version of SSL/TLS.

Misconfigured encryption: The error can also occur if the encryption parameters are not properly configured on the client or server side. For example, if the client attempts to use encryption that the server does not support or if incompatible encryption algorithms are set.

Certificate issues: Another possible reason is a problem with certificates. For instance, if the server's certificate is invalid or does not match the expected one, it can lead to an SSL/TLS error.

To resolve this error, you should consider the following steps:

Check the SSL/TLS version you are using and ensure it is compatible with the server you are trying to connect to.
Verify the encryption configuration on both the client and server sides, ensuring they are correctly set and compatible.
Check the validity and correctness of the server's certificate.
If you are using a specific library or framework, make sure it is up to date and there are no known issues related to this error.
If it's a development situation, reach out to the library or framework provider or try searching for similar issues and their solutions in online discussions or forums.

**Reply by Mato816** [reply]
We are using SMTP from Mailgun.
We are using TLS connection - TLS v1.3 (actual) or 1.2 (supported).
Can you help us with this issues?

**Comment by rdpanek** [comment]
In your version of the Lister is used `"nodemailer": "6.5.0"`

**Comment by rdpanek** [comment]
Hi @Mato816 follow pls https://github.com/canarytrace/documentation/issues/192 we work on upgrade.

**Comment by rdpanek** [comment]
Hi @Mato816 Listener was upgraded http://v2doc.canarytrace.com/docs/listener/releases#143, here is a new documentation http://v2doc.canarytrace.com/docs/listener/agent
